internal/resources/ibm/database_elasticsearch.go

```diff
@@ -1,122 +1,133 @@
-	var R decimal.Decimal
-	if r.ElasticSearch_Ram != nil {
-		R = ConvertMBtoGB(decimal.NewFromInt(*r.ElasticSearch_Ram))
-	} else { // set the default
-		R = decimal.NewFromInt(1)
-	}
-	var m decimal.Decimal
-	if r.ElasticSearch_Members != nil {
-		m = decimal.NewFromInt(*r.ElasticSearch_Members)
-	} else { // set the default
-		m = decimal.NewFromInt(3)
+func GetElasticSearchCostComponents(r *Database) []*schema.CostComponent {
+
+	if r.Flavor != "" {
+		return []*schema.CostComponent{
+			ElasticSearchHostFlavorComponent(r),
+			ElasticSearchDiskCostComponent(r),
+		}
+	} else {
+		return []*schema.CostComponent{
+			ElasticSearchRAMCostComponent(r),
+			ElasticSearchDiskCostComponent(r),
+			ElasticSearchVirtualProcessorCoreCostComponent(r),
+		}
 	}
+}
 
-	cost := R.Mul(m)
-	costComponent := schema.CostComponent{
-		Name:            "RAM",
-		Unit:            "GB-RAM",
-		MonthlyQuantity: &cost,
-		UnitMultiplier:  decimal.NewFromInt(1),
+func ElasticSearchVirtualProcessorCoreCostComponent(r *Database) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            fmt.Sprintf("Virtual Processor Cores (%s members)", strconv.FormatInt(r.Members, 10)),
+		Unit:            "CPU",
+		UnitMultiplier:  decimal.NewFromInt(1), // Final quantity for this cost component will be divided by this amount
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(r.CPU * r.Members)),
 		ProductFilter: &schema.ProductFilter{
 			VendorName:    strPtr("ibm"),
 			Region:        strPtr(r.Location),
 			Service:       strPtr("databases-for-elasticsearch"),
 			ProductFamily: strPtr("service"),
 			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "planName", Value: &r.Plan},
+				{
+					Key: "planName", Value: strPtr(r.Plan),
+				},
 			},
 		},
 		PriceFilter: &schema.PriceFilter{
-			Unit: strPtr("GIGABYTE_MONTHS_RAM"),
+			Unit: strPtr("VIRTUAL_PROCESSOR_CORES"),
 		},
 	}
-	return &costComponent
 }
 
-func ElasticSearchDiskCostComponent(r *Database) *schema.CostComponent {
-	var d decimal.Decimal
-	if r.ElasticSearch_Disk != nil {
-		d = ConvertMBtoGB(decimal.NewFromInt(*r.ElasticSearch_Disk))
-	} else { // set the default
-		d = decimal.NewFromInt(5)
+func ElasticSearchRAMCostComponent(r *Database) *schema.CostComponent {
+
+	unit := "GIGABYTE_MONTHS_RAM"
+	if r.Plan == "enterprise" {
+		unit = "GIGABYTE_MONTHS_RAM_NEW"
 	}
-	var m decimal.Decimal
-	if r.ElasticSearch_Members != nil {
-		m = decimal.NewFromInt(*r.ElasticSearch_Members)
-	} else { // set the default
-		m = decimal.NewFromInt(3)
+
+	return &schema.CostComponent{
+		Name:            fmt.Sprintf("RAM (%s members)", strconv.FormatInt(r.Members, 10)),
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),                                         // Final quantity for this cost component will be divided by this amount
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt((r.Memory * r.Members) / 1024)), // Convert to GB
```

Comment:
will there ever be a case where the value is not divisible by 1024?

Reply:
@YuChinKoo nope, the terraform only allows that value to be in increments of 1024

Reply:
Let me test it...

Reply:
* memory must be >= 1024 and <= 114688 in increments of 128
* disk must be >= 5120 and <= 4194304 in increments of 1024

Reply:
Changed it so that it will display a float instead of an int 👍🏼